tests/utils/blockchain.py

```diff
@@ -1,48 +1,116 @@
-import os
+from functools import cached_property
 from typing import List, Union
 
 import maya
+from ape.api import AccountAPI
+from ape.managers.accounts import TestAccountManager
 from eth_tester.exceptions import TransactionFailed
+from eth_typing import ChecksumAddress
 from hexbytes import HexBytes
-from web3 import Web3
 
 from nucypher.blockchain.eth.interfaces import (
     BlockchainInterface,
 )
-from nucypher.blockchain.eth.signers.software import Web3Signer
-from nucypher.blockchain.eth.token import NU
-from nucypher.crypto.powers import TransactingPower
+from nucypher.blockchain.eth.signers import Signer
+from nucypher.blockchain.eth.signers.software import InMemorySigner
 from nucypher.utilities.gas_strategies import EXPECTED_CONFIRMATION_TIME_IN_SECONDS
 from nucypher.utilities.logging import Logger
-from tests.constants import (
-    DEVELOPMENT_ETH_AIRDROP_AMOUNT,
-    INSECURE_DEVELOPMENT_PASSWORD,
-    NUMBER_OF_ETH_TEST_ACCOUNTS,
-    NUMBER_OF_STAKING_PROVIDERS_IN_BLOCKCHAIN_TESTS,
-    NUMBER_OF_URSULAS_IN_BLOCKCHAIN_TESTS,
-    TEST_ETH_PROVIDER_URI,
-)
+from tests.constants import NUMBER_OF_ETH_TEST_ACCOUNTS, TEST_ETH_PROVIDER_URI
+
+
+class ReservedTestAccountManager(TestAccountManager):
+
+    NUMBER_OF_URSULAS_IN_TESTS = 10
```

Comment:
Actually, I went down that route with this commit - https://github.com/nucypher/nucypher/pull/3491/commits/c7480e4bf5d75114de7e94e547d824ba8ce9d007, but removed the `ape-config.yml` file in a subsequent commit - https://github.com/nucypher/nucypher/pull/3491/commits/c977db21e9efd791892140dccc0c4aa81318d905. 

The reasons being:
1. It felt a bit awkward having `ape-config.yml` in the `nucypher` folder - although maybe that's just me 😅 . Of course, we can move it to the `tests` folder but if so, then the `unit` and `integration` tests would need to be run from within the `tests` folder so that the `ape-config.yml` file is picked up (analogous to what we have to do for `acceptance` tests)

2. Given that these fixtures are exported (and used by Porter), I'm not sure what effect using an `ape-config.yml` would have. Would Porter have to define its own ape-config.yml...maybe...

Agree with you that it is something interesting to explore deeper in a future PR.